ophyd/device.py

```diff
@@ -207,22 +257,33 @@ def __repr__(self):
         suffix = (repr(self.suffix) if self.suffix
                   else '')
 
-        component_class = self.cls.__name__
+        component_class = self.cls.__name__ if self.cls else ""
         args = ', '.join(s for s in (component_class, suffix, kw_str) if s)
 
         this_class = self.__class__.__name__
         return f'{this_class}({args})'
 
     __str__ = __repr__
 
-    def __get__(self, instance, owner):
+    @typing.overload
+    def __get__(self, instance: None, owner: type) -> Component:
+        ...
+
+    @typing.overload
+    def __get__(self, instance: Any, owner: type) -> K:
+        ...
+
+    def __get__(
+        self,
+        instance: Optional[Any],
+        owner: type,
+    ) -> Union[Component, K]:
         if instance is None:
             return self
 
-        try:
-            return instance._signals[self.attr]
-        except KeyError:
+        if self.attr not in instance._signals:
             return instance._instantiate_component(self.attr)
+        return instance._signals[self.attr]
```

Comment:
Is there a deeper motivation behind this inversion here? (try/except -> attr in). To me it seems purely stylistic.

Reply:
I'll revert it for the sake of an easier-to-review PR. It doesn't need to be here.

Reply:
Resolved in 1acca934a43fc8ca5577a421e9d082cb70d10966